shared/js/background/components/remote-config.js

```diff
@@ -109,19 +195,68 @@ export default class RemoteConfig extends ResourceLoader {
                     /* Handle a rollout: Dice roll is stored in settings and used that to decide
                      * whether the feature is set as 'enabled' or not.
                      */
-                    const rolloutSettingsKey = `rollouts.${featureName}.${name}.roll`;
-                    const validSteps = subfeature.rollout.steps.filter((v) => v.percent > 0 && v.percent <= 100);
-                    const rolloutPercent = validSteps.length > 0 ? validSteps.reverse()[0].percent : 0.0;
+                    const rolloutSettingsKey = getRolloutSettingsKey(featureName, name);
+                    const rolloutPercent = getSubFeatureRolloutPercent(subfeature);
                     if (!this.settings.getSetting(rolloutSettingsKey)) {
                         this.settings.updateSetting(rolloutSettingsKey, Math.random() * 100);
                     }
                     const dieRoll = this.settings.getSetting(rolloutSettingsKey);
                     subfeature.state = rolloutPercent >= dieRoll ? 'enabled' : 'disabled';
                 }
+
+                /** @type {ChosenCohort | null} */
```

Comment:
Nit: Is this `@type` necessary? I figure `getCohort()`'s return type already should let TS know `assignedCohort`'s type.